sound/soc/sof/topology.c

```diff
@@ -494,6 +494,13 @@ static int sof_control_load_volume(struct snd_soc_component *scomp,
 	}
 
 out:
+	/* set up possible led control from mixer private data */
+	scontrol->led_ctl.use_leds = 0;
+	if (le32_to_cpu(mc->priv.size) > 0) {
+		scontrol->led_ctl.use_leds = mc->priv.data[0];
+		scontrol->led_ctl.direction = mc->priv.data[1];
+	}
```

Comment:
so where would those private data come from? A topology token?
I can't recall having seen a definition for this.

Reply:
So now it is coming as private data in switch control (it is a small binary blob). As it is just 2 separate bytes it is quite easy to define even without any special tools. If it is possible to define tokens for this and add them as part of switch control's data section, I can surely do that. see related fw PR: https://github.com/thesofproject/sof/pull/1867

Reply:
probably better to make things self-explanatory with actual tokens.
binary data is error prone and obscure by construction. It should be reserved to e.g. filter coefficients or things that don't have a real meaning. for an LED enable, direction, a token is really preferable. @lgirdwood had the same comment i think of the topology PR.